src/net/alarm.rs

```diff
@@ -161,17 +155,19 @@ where
             self.subscribed = false;
         }
         if !self.subscribed && self.mqtt.client().is_connected() {
-            self.subscribed = true;
-            log::info!("Alarm subscribed to {:?}", self.prefix);
-            self.mqtt
+            let result = self
+                .mqtt
                 .client()
                 .subscribe(&[(self.prefix.as_str()).into()], &[])
-                .map(|_| None) // return None if we just subscribed
+                .map(|_| None); // return None if we just subscribed
+            self.subscribed = true;
+            log::info!("Alarm subscribed to {:?}", self.prefix);
+            result
         } else {
             self.mqtt
-                .poll(|_client, _topic, message, _properties| match message {
-                    TRUE => Some(true),
-                    FALSE => Some(false),
+                .poll(|_client, _topic, message, _properties| match from_slice(message) {
+                    Ok((true, 4)) => Some(true),
+                    Ok((false, 5)) => Some(false),
```

Comment:
```suggestion
                    Ok((b, _)) => Some(b),
```

And then clippy might be even wiser.